src/kernel.ts

```diff
@@ -364,7 +372,10 @@ class Kernel {
       throw Error('You must set the kernel id before starting.');
     }
     this._handleStatus('starting');
-    return utils.ajaxRequest(this._kernelUrl, {
+    if (this._promiseDelegate.settled) {
```

Comment:
Same comment, having to check whether the promise is settled doesn't seem right for some reason. I want to make sure we have a full understanding of what's going on, and the potential issues this could cause in the present of multiple in-flight requests.